Doc/library/typing.rst

```diff
@@ -1940,6 +1940,25 @@ Other concrete types
    represent the types of I/O streams such as returned by
    :func:`open`.
 
+   These types should be used sparingly. For argument types, a tight
+   :class:`Protocol` that documents the actual needs of the implementation
+   is often a better alternative. Return values should be annotated with
+   the actual concrete type returned. For example::
+
+       from io import StringIO
+
+       class Reader(Protocol):
+           def read(self, __size: int) -> bytes: ...
```

Comment:
```suggestion
           def read(self, size: int, /) -> bytes: ...
```

Reply:
Oh, how I wish we could use that in stubs already. :)